articles/performance-optimization-techniques-for-node-js/index.md

```diff
@@ -0,0 +1,376 @@
+
+### Gzip Compression
+Gzip compresses HTTP requests and responses. [Gzip](https://nginx.org/en/docs/http/ngx_http_gzip_module.html) compresses responses before sending them to the browser, thus, the browser takes a shorter time to fetch them. Gzip also compresses the request to the remote server, which significantly increases web performance.
+
+When using [Express.js](http://expressjs.com/), choose to use [compression](https://github.com/expressjs/compression). It is a Node.js compression middleware that supports deflate, Gzip, and serve static content. The compression library  will compress every request that passes through the middleware.
+
+Example
+
+```js
+const compression = require("compression");
+const express = require("express");
+const app = express();
+// compress all responses
+app.use(compression());
+// add all routes
+```
+
+"Here are more details on how to [connect express with compression](https://github.com/expressjs/compression#examples).
+
+### Avoid Memory Leaks
+A [memory leak](https://en.wikipedia.org/wiki/Memory_leak) occurs when a computer program incorrectly manages memory allocations in a way that memory which is no longer needed is not released. With memory leaks, a loaded page will reserve more and more memory. These will slowly occupy your available space to a point where CPU usage is overstretched. Such scenarios will severely affect your application performance.
+
+You may choose to restart your application and these issues will magically disappear but this will not be a reliable solution. The problem will repeat itself and the memory will pile up periodically. Memory leaks can come from a [DOM](/engineering-education/document-object-model/) that has been removed from a page but some variables still make some reference to them. If you happen to remove [DOM](/engineering-education/document-object-model/) values, make sure to set them to null. When null, the garbage collector will eliminate them and avoid memory leaks.
+
+Use [Chrome Dev Tools](https://developers.google.com/web/tools/chrome-devtools?utm_source=dcc&amp;utm_medium=redirect&amp;utm_campaign=2018Q2) to analyze if your web page has any memory leaks. Chrome Dev Tools will help you catch and debug memory leaks. [This article](https://marmelab.com/blog/2018/04/03/how-to-track-and-fix-memory-leak-with-nodejs.html) will help you understand more about how to detect and solve memory leaks using Chrome Dev Tools.
+
+### Real-time Monitoring
+Monitoring helps get insights into your production application to ensure a stable and reliable system. Insights are critical in helping detect performance problems within your system.
```

Comment:
Instead of saying "Insights are critical", try something like "insight is critical"